### ISSUE TITLE
per-monitor dpi awareness on windows

### ISSUE THREAD

**Issue by valignatev** [akiyosi/goneovim#315]
Hi! This is just a feature request to support dragging goneovim between monitors with different dpi. I have one monitor with 1440p resolution, and the other one with 2160p. When I drag the program from the first to the second one, it becomes blurry as if hidpi wasn't supported at all. neovim-qt works well though. Most likely, goneovim isn't aware about a dpi change because it doesn't use per-monitor v2 awareness mode: https://docs.microsoft.com/en-us/windows/win32/hidpi/dpi-awareness-context

**Comment by akiyosi** [comment]
@valignatev 
hi, Thanks for this report.

Just to confirm, do the following settings change the behavior of this issue?

```
# Put the following in setting
Cacheddrawing = false
```

If you are using Windows, place the setting in the following location:

```
~\.goneovim\settings.toml
```

**Comment by valignatev** [comment]
no, this setting doesn't affect anything. This is 90% per-monitor dpi awareness issue, because the scaling works if you initially launch goneovim on the second monitor. So the only issue is detecting the dpi change upon dragging over the monitor boundary

**Comment by akiyosi** [comment]
@valignatev 
Thanks for the confirmation!

I've created a trial fix for this issue.
Could you please check the reproducibility of the issue with the binary generated by CI as follows?
(I do not have different DPI displays and it is difficult to check right away...)

https://github.com/akiyosi/goneovim/actions/runs/1796089670

**Comment by valignatev** [comment]
Hey, thanks for getting onto it! I've tried the job artifact, and now it's not blurry anymore, but the ui doesn't scale properly - so if you launch the program on a hidpi monitor, or if you drag it there everything is very small, because there's no scaling apparently. An expected behavior is that upon launching/dragging the editor to the monitor with different dpi (whether it's higher or lower) there's some mechanism in place that scales all of the UI in a way that it appears to be the desired size and isn't blurry at the same time.

Unfortunately, I don't know how to do that in Qt and how it wraps windows apis, so can't help much. A documentation on QT_AUTO_SCREEN_SCALE_FACTOR says something about point-sized fonts not being scaled, but I'm not sure if it's related or not

**Comment by valignatev** [comment]
This introduced blurry scaling again

**Comment by akiyosi** [comment]
@valignatev 
Thank you.
By the way, in the above test artifact, does the behavior change depending on whether the following goneovim config setting is enabled or disabled?

```
# Put the following in ~\.goneovim\settings.toml
[Editor]
Cacheddrawing = false
```

The reason I am concerned about this setting is that when it is enabled, Goneovim stores images internally as a cache of drawn text. I was wondering if images have anything to do with dpi awareness.

**Comment by valignatev** [comment]
no, not quite. In that neovim-qt issue, the letters got scaled properly, but the spacing got screwed. In goneovim's case, the scaling seemed to just do bitmap stretching resulting in blur. Here you can see goneovim and neovim-qt opened side by side after dragging them from 96 to 200+ dpi monitor:

![image](https://user-images.githubusercontent.com/7038954/153232242-40ab12b4-8017-41a3-826e-debe88f9b856.png)

I hope the difference in how sharp the the font is is noticeable on the picture

**Comment by akiyosi** [comment]
Thanks for the screenshots! It's very helpful.
Is there any change in the behavior of the following CI artifacts?

https://github.com/akiyosi/goneovim/actions/runs/1809691548
